tests/Aql/ShortestPathExecutorTest.cpp

```diff
@@ -0,0 +1,434 @@
+  }
+
+  std::vector<std::string> const& findPath(std::pair<std::string, std::string> const& src) {
+    for (auto const& p : _paths) {
+      if (p.front() == src.first && p.back() == src.second) {
+        return p;
+      }
+    }
+    REQUIRE(false);
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
+  }
+
+  std::pair<std::string, std::string> const& calledAt(size_t index) {
+    REQUIRE(index < _calledWith.size());
+    return _calledWith[index];
+  }
+
+  // Needs to provide lookupFunctionality for Cache
+ private:
+  std::vector<std::vector<std::string>> _paths;
+  std::vector<std::pair<std::string, std::string>> _calledWith;
+  TokenTranslator& _translator;
+};
+
+struct TestShortestPathOptions : public ShortestPathOptions {
+  TestShortestPathOptions(Query* query) : ShortestPathOptions(query) {
+    std::unique_ptr<TraverserCache> cache = std::make_unique<TokenTranslator>(query);
+    injectTestCache(std::move(cache));
+  }
+};
+
+static void ValidateResult(ShortestPathExecutorInfos& infos, OutputAqlItemRow& result,
+                           std::vector<std::pair<std::string, std::string>> const& resultPaths) {
+  if (!resultPaths.empty()) {
+    FakePathFinder& finder = static_cast<FakePathFinder&>(infos.finder());
+    THEN("validate output") {
+      TokenTranslator& translator = *(static_cast<TokenTranslator*>(infos.cache()));
+      auto block = result.stealBlock();
+      REQUIRE(block != nullptr);
+      size_t index = 0;
+      for (size_t i = 0; i < resultPaths.size(); ++i) {
+        auto path = finder.findPath(resultPaths[i]);
+        for (size_t j = 0; j < path.size(); ++j) {
+          if (infos.usesOutputRegister(ShortestPathExecutorInfos::VERTEX)) {
+            AqlValue value =
+                block->getValue(index, infos.getOutputRegister(ShortestPathExecutorInfos::VERTEX));
+            CHECK(value.isObject());
+            CHECK(arangodb::basics::VelocyPackHelper::compare(
+                      value.slice(),
+                      translator.translateVertex(StringRef(path[j])), false) == 0);
+          }
+          if (infos.usesOutputRegister(ShortestPathExecutorInfos::EDGE)) {
+            AqlValue value =
+                block->getValue(index, infos.getOutputRegister(ShortestPathExecutorInfos::EDGE));
+            if (j == 0) {
+              CHECK(value.isNull(false));
+            } else {
+              CHECK(value.isObject());
+              VPackSlice edge = value.slice();
+              // FROM and TO checks are enough here.
+              CHECK(StringRef(edge.get(StaticStrings::FromString)).compare(path[j - 1]) == 0);
+              CHECK(StringRef(edge.get(StaticStrings::ToString)).compare(path[j]) == 0);
+            }
+          }
+          ++index;
+        }
+      }
+    }
+  }
+}
+static void TestExecutor(ShortestPathExecutorInfos& infos,
+                         std::shared_ptr<VPackBuilder> input,
+                         std::vector<std::pair<std::string, std::string>> const& resultPaths) {
```

Comment:
Why `TestExecutor` instead of `testExecutor`? I've actually mistaken it for a constructor on the first glance. :wink: